init.lua

```diff
@@ -142,9 +172,38 @@ function SkyRocket:handleDrag()
       })
 
       return true
-    else
-      return nil
     end
+
+    if self:isFancy() then
+      self.fancyPosition.x = self.fancyPosition.x + dx
+      self.fancyPosition.y = self.fancyPosition.y + dy
+      if self.fancyPosition.x < 0 then self.fancyPosition.x = 0 else if self.fancyPosition.x > self.screen.w then self.fancyPosition.x = self.screen.w end end
+      if self.fancyPosition.y < 0 then self.fancyPosition.y = 0 else if self.fancyPosition.y > self.screen.h then self.fancyPosition.y = self.screen.h end end
+
+      for k,v in pairs(self.zones) do            
+          if 
+              (self.fancyPosition.x > v.x*self.screen.w) and 
```

Comment:
Done and also instead of transforming the zone ratios to absolute position I changed reversed the operations and now change the absolute position to a ratio. Instead of performing n operations (n = number of zones), now only one is performed.